dpnp/backend/extensions/statistics/sliding_dot_product1d.cpp

```diff
@@ -0,0 +1,170 @@
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+#include <algorithm>
+#include <complex>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <vector>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+// dpctl tensor headers
+#include "dpctl4pybind11.hpp"
+#include "utils/type_dispatch.hpp"
+
+#include "common.hpp"
+#include "sliding_dot_product1d.hpp"
+#include "sliding_window1d.hpp"
+
+#include <iostream>
+
+namespace dpctl_td_ns = dpctl::tensor::type_dispatch;
+using dpctl::tensor::usm_ndarray;
+
+using namespace statistics::sliding_window1d;
+using namespace statistics::common;
+
+namespace
+{
+
+template <typename T>
+struct SlidingDotProductF
+{
+    static sycl::event impl(sycl::queue &exec_q,
+                            const void *v_ain,
+                            const void *v_vin,
+                            void *v_out,
+                            const size_t a_size,
+                            const size_t v_size,
+                            const size_t l_pad,
+                            const size_t r_pad,
+                            const std::vector<sycl::event> &depends)
+    {
+        const T *ain = static_cast<const T *>(v_ain);
+        const T *vin = static_cast<const T *>(v_vin);
+        T *out = static_cast<T *>(v_out);
+
+        auto device = exec_q.get_device();
+        const auto local_size = get_max_local_size(device);
+        const auto work_size = l_pad + r_pad + a_size - v_size + 1;
+
+        constexpr int32_t WorkPI = 4;
+
+        return exec_q.submit([&](sycl::handler &cgh) {
+            cgh.depends_on(depends);
+
+            auto input = make_padded_span(ain, a_size, l_pad);
+            auto kernel = make_span(vin, v_size);
+            auto result = make_span(out, work_size);
+
+            if (v_size > 8) {
+                const auto nd_range =
+                    make_ndrange(work_size, local_size, WorkPI);
+                submit_sliding_window1d<WorkPI>(
+                    input, kernel, std::multiplies<>{}, std::plus<>{}, result,
+                    nd_range, cgh);
+            }
+            else {
+                const auto nd_range =
+                    make_ndrange(work_size, local_size, WorkPI);
+                submit_sliding_window1d_small_kernel<WorkPI>(
+                    input, kernel, std::multiplies<>{}, std::plus<>{}, result,
+                    nd_range, cgh);
+            }
+        });
+    }
+};
+
+using SupportedTypes = std::tuple<uint64_t,
```

Comment:
Would it make sense to add more integer dtypes?

Reply:
I'm trying to have as less supported types as possible. Though having `int32` and `uint32` could be useful